File: utilities/serverClient.js
const { Client, GatewayIntentBits, Partials } = require("discord.js");

let client = new Client({
  intents: [
    GatewayIntentBits.Guilds,
    GatewayIntentBits.GuildMessages,
    GatewayIntentBits.MessageContent,
    GatewayIntentBits.GuildMembers,
  ],
  partials: [Partials.Message, Partials.Channel],
});

module.exports = {
  loginClientToDiscord: async () => {
    await client.login(process.env.DISCORD_TOKEN);
  },
  getClient: () => client,
};

File: src/utilities/server_client.rs
use serenity::{all::{ClientBuilder, GatewayIntents}, Client, Result, client::EventHandler, async_trait};
use std::env;
use dotenv::dotenv;
use serenity::prelude::*;
use serenity::model::gateway::Ready;

struct Handler;

#[async_trait]
impl EventHandler for Handler {
    async fn ready(&self, _: Context, ready: Ready) {
        println!("{} is online!", ready.user.name)
    }
}

pub async fn get_client() -> Result<Client> {
    dotenv().ok();
    let token: String = env::var("DISCORD_TOKEN").expect("Expected a discord token");

    Ok(Client::builder(token, GatewayIntents::GUILD_MESSAGES | GatewayIntents::MESSAGE_CONTENT | GatewayIntents::GUILD_MEMBERS).event_handler(Handler).await.expect("Error creating client"))
}